integration/pages/BaseTestPage.js

```diff
@@ -0,0 +1,68 @@
+const BasePage = require('./BasePage');
+const assert = require('assert');
+const { By } = require('selenium-webdriver');
+
+const ELEMENTS = {
+  appHeader: By.css('[data-testid=\'app-name\']'),
+  meetingIdInputBox: By.css('[data-testid=\'input\'][name=\'Meeting Id\']'),
+  nameInputBox: By.css('[data-testid=\'input\'][name=\'Name\']'),
+  joinMeetingButton: By.css('[data-testid=\'button\'][label=\'Join Meeting\']'),
+  leaveMeetingButton: By.css('[data-testid=\'button\'][label=\'Leave\']'),
+  meetingStatus: By.id('meeting-status'),
+};
+
+class BaseTestPage extends BasePage {
+  constructor(driver) {
+    super(driver);
+  }
+
+  async load(route) {
+    await this.visit(route);
+  }
+
+  async enterMeetingInfo(meetingId, name) {
+    const meetingIdInputBox = this.find(ELEMENTS.meetingIdInputBox);
+    await meetingIdInputBox.sendKeys(meetingId);
+
+    const nameInputBox = this.find(ELEMENTS.nameInputBox);
+    await nameInputBox.sendKeys(name);
+  }
+
+  async joinMeeting() {
+    const joinMeetingButton = this.find(ELEMENTS.joinMeetingButton);
+    await joinMeetingButton.click();
+  }
+
+  async leaveMeeting() {
+    const leaveMeetingButton = this.find(ELEMENTS.leaveMeetingButton);
+    await leaveMeetingButton.click();
+  }
+
+  async checkIfAttendeeHasJoinedMeeting() {
+    const isMatched = await this.waitUntil(async () => {
+      const meetingStatus = await this.find(ELEMENTS.meetingStatus).getText();
+      return meetingStatus === 'Meeting Status: Succeeded';
+    }, 10000);
+
+    assert(isMatched, 'Attendee did not join the meeting successfully');
+  }
+
+  async checkIfAppLoaded() {
+    const isDisplayed = await this.isDisplayed(
+      ELEMENTS.appHeader,
+      10000
+    );
+    assert(isDisplayed, 'App was not loaded successfully');
+  }
+
+  async checkIfAttendeeHasLeftMeeting() {
+    const isMatched = await this.waitUntil(async () => {
+      const meetingStatus = await this.find(ELEMENTS.meetingStatus).getText();
+      return meetingStatus === 'Meeting Status: Left';
+    }, 10000);
```

Comment:
10s is really long. Can it be shorter?

Reply:
This is not a fixed wait time. It means it will wait until the change appears but no longer than 10s. I don't have enough data to choose a proper timeout, so make it 10s. In my local test, it usually need 0.3s, but not sure how it goes when running on Sauce Labs.
Do you have any suggestions?

Reply:
Yeah i got that but 10s is really long. Can we just set it to 5 for now? If it takes longer than 5s then we probably want to fail the test :D